libc/calls/commandv.c

```diff
@@ -57,6 +59,10 @@ static bool AccessCommand(const char *name, char *path, size_t pathsz,
   }
   memcpy(path + pathlen, name, namelen);
   memcpy(path + pathlen + namelen, suffix, suffixlen + 1);
+  if (IsWindows() &&
+          ((*suffix == '\0' && memchr(basename(name), '.', namelen) == NULL)) ||
```

Comment:
Rather then determining the basename and searching the string. Why not compare with directly what we want:
```c
((suffixlen == 0) && !IsExePath(name, namelen) && !IsComPath(name, namelen))
```
Minor nitpick, checking suffixlen is a better test of an empty string as it doesn't require dereferencing a pointer first.